### DISCUSSION TITLE
How to convert .txt file to MATLAB file?

### DISCUSSION THREAD

**Discussion by VeerammalV** [community/community#143138] [Programming Help]
### Body

How can I convert a .txt file into a MATLAB script file (with a .m extension) without using MATLAB and ensure the content is properly formatted as MATLAB code?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SyedAliJunaidSherazi** [comment]
Hi there!

To convert a `.txt` file into a MATLAB script file with the `.m` extension, you can follow these steps:

1. **Open the .txt File**: Start by opening your `.txt` file in a text editor or directly in MATLAB. Make sure the content you want to convert is properly formatted as MATLAB code.

2. **Check MATLAB Syntax**: Review the content to ensure it adheres to MATLAB's syntax rules:
   - Make sure functions are defined properly.
   - Check that variables are assigned correctly.
   - Comments should start with a `%` symbol.
   - Remove any unnecessary formatting or characters that may not be recognized by MATLAB.

3. **Save as .m File**:
   - **If you're using MATLAB**:
     - Go to the `File` menu and select `Save As`.
     - Change the file type to "MATLAB Files (*.m)".
     - Enter your desired script name and click `Save`.
   - **If you're using a text editor**:
     - Select `Save As`.
     - Change the file extension from `.txt` to `.m`, and ensure the file type is set to "All Files" if necessary.

4. **Test Your Script**: Finally, open the newly created `.m` file in MATLAB and run it to ensure it executes without errors. This will confirm that everything is formatted correctly.

If you run into any issues or have further questions, feel free to ask! **And if this answer helps resolve your problem, I’d appreciate it if you could mark my comment as answered.** I'm here to help!

**Comment by divsansi** [comment]
response #23515

**Comment by nabinghosh** [comment]
Hey everyone! 👋

I wanted to share a simple way to convert text files to MATLAB format. Here's a straightforward approach that works well:

```matlab
% Method 1: Using dlmread (great for numeric data)
% Assuming your txt file has comma-separated values
data = dlmread('input.txt', ',', 1, 0);  % Skip header row if present
save('output.mat', 'data');

% Method 2: Using readmatrix (newer MATLAB versions)
data = readmatrix('input.txt');
save('output.mat', 'data');
